### PR TITLE
[Mellanox] Add support for SN2201 platform

### PR DESCRIPTION
**Pull request by liorghub**
Signed-off-by: liora <liora@nvidia.com>

#### Why I did it
Add support for SN2201 platform

#### How I did it
Add required content for SN2201 platform
Note: still missing kernel driver support for this system. Once all is upstream will be updated as well.

#### How to verify it
Install and basic sanity tests including traffic. 

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

**Comment by liorghub** [comment]
Build failed due to timeout of kvm-t0 test.
Timeout value is increased now (https://github.com/Azure/sonic-buildimage/pull/9336).
@lguohan can you rerun please?

File: platform/mellanox/asic_table.j2 [hunk 1]
```diff
@@ -40,6 +40,7 @@
         'x86_64-mlnx_msn4700-r0':'MELLANOX-SPECTRUM-3',
         'x86_64-mlnx_msn4600c-r0':'MELLANOX-SPECTRUM-3',
         'x86_64-mlnx_msn4600-r0':'MELLANOX-SPECTRUM-3',
+        'x86_64-nvidia_sn2201-r0': 'MELLANOX-SPECTRUM',
```
Comment (qiluo-msft) [inline platform/mellanox/asic_table.j2:43]: >  [](http://example.com/codeflow?start=34&length=1)

Minor issue: remove extra blank

**Comment by qiluo-msft** [comment]
You change more code than 2201. Would you like to explain in title and description?

File: platform/mellanox/asic_table.j2 (hunk 1 above)
Comment (liorghub) [reply platform/mellanox/asic_table.j2:43]: I couldn't find any whitespace to remove, am I missing something?

**Comment by liorghub** [comment]
> You change more code than 2201. Would you like to explain in title and description?

Actually I changed code only for 2201. 
Some files are soft link to another file that resides on another system folder, could that be the reason for your comment?

**Review by liat-grozovik — DISMISSED** [review]

**Comment by liat-grozovik** [comment]
@qiluo-msft any further comment we should address? If not I would like to have this system part of 202111

File: platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py [hunk 2]
```diff
@@ -153,7 +153,14 @@
...
         }
-    }
+    },
+    'x86_64-nvidia_sn2201-r0': {
+        'thermal': {
+            "capability": {
+                "comex_amb": False
+            }
+        },
+    },
```
Comment (liat-grozovik) [inline platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py:163]: not sure comma is needed here.

File: device/mellanox/x86_64-nvidia_sn2201-r0/pmon_daemon_control.json [hunk 3]
```diff
@@ -0,0 +1 @@
+../x86_64-mlnx_msn2700-r0/pmon_daemon_control.json
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-nvidia_sn2201-r0/pmon_daemon_control.json:1]: please check path. might be that you need extra ../ as in other links

File: device/mellanox/x86_64-nvidia_sn2201-r0/system_health_monitoring_config.json [hunk 4]
```diff
@@ -0,0 +1 @@
+../x86_64-mlnx_msn2700-r0/system_health_monitoring_config.json
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-nvidia_sn2201-r0/system_health_monitoring_config.json:1]: please check path. might be that you need extra ../ as in other links

File: device/mellanox/x86_64-nvidia_sn2201-r0/thermal_policy.json [hunk 5]
```diff
@@ -0,0 +1 @@
+../x86_64-mlnx_msn2700-r0/thermal_policy.json
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-nvidia_sn2201-r0/thermal_policy.json:1]: please check path. might be that you need extra ../ as in other links

**Review by liat-grozovik — COMMENTED** [review]
as commented

File: platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py (hunk 2 above)
Comment (liorghub) [reply platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py]: Actually it is harmless but I removed it to keep the convention.

File: device/mellanox/x86_64-nvidia_sn2201-r0/pmon_daemon_control.json (hunk 3 above)
Comment (liorghub) [reply device/mellanox/x86_64-nvidia_sn2201-r0/pmon_daemon_control.json:1]: I verified and the pah is valid. 
The red color disturbed me so I looked on other systems that have soft link and it is red there as well.
see for example: https://github.com/Azure/sonic-buildimage/blob/bf816964137444a1f1145656c2d4b60254620a8f/device/mellanox/x86_64-mlnx_msn3420-r0/thermal_policy.json

File: device/mellanox/x86_64-nvidia_sn2201-r0/system_health_monitoring_config.json (hunk 4 above)
Comment (liorghub) [reply device/mellanox/x86_64-nvidia_sn2201-r0/system_health_monitoring_config.json:1]: I verified and the path is valid.

File: device/mellanox/x86_64-nvidia_sn2201-r0/thermal_policy.json (hunk 5 above)
Comment (liorghub) [reply device/mellanox/x86_64-nvidia_sn2201-r0/thermal_policy.json:1]: I verified, the path is valid.

**Review by liat-grozovik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@liorghub could you please change the default SKU to be without the M? Like SN2201 and not MSN5600

**Comment by liorghub** [comment]
> @liorghub could you please change the default SKU to be without the M? Like SN2201 and not MSN5600

@liat-grozovik done, thanks for the catch.

File: platform/mellanox/asic_table.j2 (hunk 1 above)
Comment (liorghub) [reply platform/mellanox/asic_table.j2]: @qiluo-msft I removed the whitespace.

**Review by qiluo-msft — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**